cmd/doctor.go

```diff
@@ -0,0 +1,92 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+
+	"github.com/spf13/cobra"
+	vfs "github.com/twpayne/go-vfs"
+)
+
+// doctorCmd represents the doctor command
+var doctorCommand = &cobra.Command{
+	Use:   "doctor",
+	Short: "Check your system for potential problems",
+	Long: `Check your system for potential problems. Such as:
+* Does the configuration file exist
+* Does the destination directory exist
+* Does the source directory exist
+* Is Hashicorp Vault installed?
+* Is Bitwarden installed?
+* Is LastPass installed?
+`,
+	RunE: makeRunE(config.runDoctorCommand),
+}
+
+func init() {
+	rootCommand.AddCommand(doctorCommand)
+
+}
+
+func (c *Config) runDoctorCommand(fs vfs.FS, args []string) error {
+	fmt.Printf("Arch: %s\n", runtime.GOARCH)
+	fmt.Printf("OS: %s\n", runtime.GOOS)
+
+	info, err := os.Stat(configFile)
+	if os.IsNotExist(err) {
+		fmt.Printf("Configuration File: Does Not Exist (%s)\n", configFile)
+	} else {
+		fmt.Printf("Configuration File: Exists (%s %s)\n", info.Mode().Perm().String(), configFile)
+	}
+
+	info, err = os.Stat(config.DestDir)
+	if os.IsNotExist(err) {
+		fmt.Printf("Destination Directory: Does Not Exist (%s)\n", config.DestDir)
+	} else {
+		fmt.Printf("Destination Directory: Exists (%s %s)\n", info.Mode().Perm().String(), config.DestDir)
+	}
+
+	info, err = os.Stat(config.SourceDir)
+	if os.IsNotExist(err) {
+		fmt.Printf("Source Directory: Does Not Exist (%s)\n", config.SourceDir)
+	} else {
+		fmt.Printf("Source Directory: Exists (%s %s)\n", info.Mode().Perm().String(), config.SourceDir)
+	}
```

Comment:
Please refactor the `config.DestDir` and `config.SourceDir` tests into a `for`-loop over a slice, as described for the binaries.

Reply:
Updated